src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -487,6 +491,24 @@ public function firstOrCreate(array $attributes = [], array $values = [])
         });
     }
 
+    /**
+     * Find a model by its primary key. If not found, creates a new record.
+     *
+     * @param  mixed  $id
+     * @param  array  $values
+     * @return \Illuminate\Database\Eloquent\Model|static
+     */
+    public function findOrCreate($id, array $values = [])
```

Comment:
I don't really see a use case for this. The `firstOrCreate` method is useful because it searches models for any field you want, and if it doesn't exist, a new model will be created with the values ​​you searched for merged with the second array. Since you're not merging id into `$values`, which wouldn't even make sense since `id` is generated at the database level, why would you need to check if any models exist by their id and then create a new one with values ​​that you didn't even use it to check if the model exists in the first place?

Reply:
Previously
`$user = User::firstOrcreate(['id'=>28], ['name'=>'navaneeth', 'age'=>27]);`

Now
`$user = User::findOrcreate(28, ['name'=>'navaneeth', 'age'=>27]);`

2 benefits
1. No need to type key name.
2. Just in case if key is null then no need to run a DB query.

Reply:
@mateusjunges If you still think it is useless, what is `findOrNew()` doing in the first place 😏